uiplib/gui/gallery.py

```diff
@@ -42,7 +44,12 @@ def set_image(self, imagePath):
                                 label="Blur", showvalue=0)
             self.slider.pack()
         self.image = UipImage(imagePath)
-        self.show_image(self.image)
+        im = Image.open(imagePath)
+        try:
+            im.load()
```

Comment:
Are we sure this is how we want to do this? Maybe add this to UIP image object? Consult @abhsag24